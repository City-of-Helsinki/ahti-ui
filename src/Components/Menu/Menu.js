import styled from 'styled-components';

const Menu = styled.nav`
  box-sizing: border-box;

  z-index: 1337;
  position: absolute;
  position: fixed;

  display: flex;

  /* NOTE: This is the arbitrary max-width for wide screens */
  max-width: 474px;

  /* NOTE: This height matters for overlapping the map  */
  min-height: 8vh;
  width: 100%;
  padding: 2rem 1.75rem;

  justify-content: space-between;
  align-items: center;

  background-color: ${props => props.theme.colors.white};
  box-shadow: ${props => props.theme.shadows.menuShadow};

  .mainTitle {
    display: block;
    height: 100%;
    margin: 0;
    font-size: 2rem;
    color: ${props => props.theme.colors.black};
  }

  button {
    display: inline-block;
  }
`;

export default Menu;
