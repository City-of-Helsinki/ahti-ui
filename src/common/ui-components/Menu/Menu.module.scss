@import '../../../styles/icons';
@import '../../../styles/partials/variables';
@import '../../../styles/partials/mixins';

$menu-z-index: 1000;
.container {
  width: 100%;
  padding: $padding;
  height: $menu_height;
  position: absolute;
  top: 0;
  left: 0;
  background-color: transparent;
  z-index: $menu-z-index;
}

.container_open {
  padding: $padding;
  height: 100vh;
  width: 100%;

  position: absolute;
  top: 0;
  background-color: var(--hds-ui-color-white);
  z-index: $menu-z-index;
}

.search {
  width: $search-component-width;
}

.searchResults {
  width: $search-component-width;
  z-index: $menu-z-index;
}

.headerContainer {
  @include flex;
  justify-content: space-between;
  padding-top: $padding;
  padding-bottom: $padding;
  // border-bottom: 1px solid var(--hds-ui-color-black-40);
  background: transparent;
}

.navLink {
  text-decoration: none;
  line-height: $line-height;
  color: var(--hds-ui-color-black);
}

.navLinkDisabled {
  @include disabledLink;
  color: var(--hds-ui-color-black-40);
}

.navDropdownContent {
  @include flex(true);
}
