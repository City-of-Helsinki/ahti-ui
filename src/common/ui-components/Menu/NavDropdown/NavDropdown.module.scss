@import '../../../../styles/icons';

.container {
  display: grid;
  grid-template-columns: 56px auto $icon-width;
  grid-row-gap: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  border-bottom: 1px solid var(--hds-ui-color-black-40);
  background-color: var(--hds-ui-color-white);
}

.headerContainer {
  display: grid;
  grid-template-columns: 56px auto $icon-width;
  grid-column: 1 / -1;
  cursor: pointer;
  color: var(--hds-ui-color-black);
}

.titleContainer {
  font-weight: 700;
}

.childrenContainer {
  grid-column: 2 / 2;
}

.angleIconOpen {
  transform: rotate(-90deg);
}

.angleIconClosed {
  transform: rotate(90deg);
}
