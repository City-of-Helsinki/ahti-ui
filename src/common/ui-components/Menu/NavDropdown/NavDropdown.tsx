import React, { useState } from 'react';
import classNames from 'classnames/bind';
import { IconAngleRight } from 'hds-react';
import { useTranslation } from 'react-i18next';

import styles from './NavDropdown.module.scss';
import CategoryIcon from '../../CategoryIcon/CategoryIcon';

const cx = classNames.bind(styles);

export interface NavDropdownProps {
  readonly title: string;
  readonly category: string;
  readonly initiallyOpen?: boolean;
}

const NavDropdown: React.FC<NavDropdownProps> = ({
  title,
  category,
  initiallyOpen = false,
  children
}) => {
  const [isOpen, setIsOpen] = useState(initiallyOpen);
  const { t } = useTranslation();

  // FIXME: layout:grid doesn't work inside buttons on chrome. hence a div is used here instead
  return (
    <div className={styles.container}>
      <div
        className={styles.headerContainer}
        onClick={() => setIsOpen(!isOpen)}
        role={'button'}
        aria-pressed={isOpen}
        aria-label={isOpen ? t('menu.close_submenu') : t('menu.open_submenu')}
      >
        <CategoryIcon category={category} className={styles.icon} />
        <div className={styles.titleContainer}>{title}</div>
        <IconAngleRight
          className={cx({
            icon: true,
            angleIconOpen: isOpen,
            angleIconClosed: !isOpen
          })}
        />
      </div>
      {isOpen && <div className={styles.childrenContainer}>{children}</div>}
    </div>
  );
};

export default NavDropdown;
