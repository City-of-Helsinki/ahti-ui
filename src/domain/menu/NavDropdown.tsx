import React, { useState } from 'react';
import classNames from 'classnames/bind';
import styles from './NavDropdown.module.scss';
import { IconAngleRight } from 'hds-react';

const cx = classNames.bind(styles);

interface NavDropdownProps {
  readonly title: string;
  readonly icon: React.ReactElement;
  readonly initiallyOpen?: boolean;
}

const NavDropdown: React.FC<NavDropdownProps> = ({
  title,
  icon,
  initiallyOpen = false,
  children,
}) => {
  const [isOpen, setIsOpen] = useState(initiallyOpen);

  return (
    <div className={styles.container}>
      <div
        className={styles.headerContainer}
        onClick={() => setIsOpen(!isOpen)}
      >
        {React.cloneElement(icon, { className: styles.icon })}
        <h1>{title}</h1>
        <IconAngleRight
          className={cx({
            icon: true,
            angleIconOpen: isOpen,
            angleIconClosed: !isOpen,
          })}
        />
      </div>
      {isOpen && <div className={styles.childrenContainer}>{children}</div>}
    </div>
  );
};

export default NavDropdown;
