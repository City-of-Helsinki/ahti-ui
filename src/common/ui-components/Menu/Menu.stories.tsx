import React from 'react';
import Menu from './Menu';
import { IconLocation, IconFood } from 'hds-react';
import LanguageSelect from '../LanguageSelect/LanguageSelect';
import { SUPPORTED_LANGUAGES } from '../../translation/TranslationConstants';
import { BrowserRouter as Router } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

const defaultMenuItems = [
  {
    title: 'islands',
    icon: <IconLocation />,
    links: [{ name: 'all_islands', url: '/map?type=myhelsinki' }]
  },
  {
    title: 'travel_to_islands',
    icon: <IconLocation />,
    links: [
      { name: 'water_bus_and_ferries', url: '/', disabled: true },
      { name: 'boat_rides_and_taxis', url: '/', disabled: true },
      { name: 'sightseeing_and_cruises', url: '/', disabled: true }
    ]
  },
  {
    title: 'saunas_and_swimming',
    icon: <IconLocation />,
    links: [
      { name: 'saunas', url: '/', disabled: true },
      { name: 'swimming', url: '/', disabled: true },
      { name: 'ice_swimming', url: '/', disabled: true }
    ]
  },
  {
    title: 'restaurants',
    icon: <IconFood />,
    links: [
      { name: 'restaurants', url: '/', disabled: true },
      { name: 'cafes', url: '/', disabled: true },
      { name: 'beer_and_wine_bars', url: '/', disabled: true }
    ]
  },
  {
    title: 'boater_services',
    icon: <IconLocation />,
    links: [
      { name: 'mooring', url: '/', disabled: true },
      { name: 'maintenance', url: '/', disabled: true },
      { name: 'septic_tank_draining', url: '/', disabled: true }
    ]
  },
  {
    title: 'rental_boats',
    icon: <IconLocation />,
    links: [
      { name: 'rowboats', url: '/', disabled: true },
      { name: 'motorboats', url: '/', disabled: true },
      { name: 'other', url: '/', disabled: true }
    ]
  }
];

export default {
  title: 'Menu',
  decorators: [(storyFn: () => React.ReactNode) => <Router>{storyFn()}</Router>]
};

export const TranslatedMenuItems = () => {
  return <Menu menuItems={defaultMenuItems} translate={true} />;
};

TranslatedMenuItems.story = {
  name: 'Translated menu items'
};

export const MenuWithLanguageSelect = () => {
  return (
    <Menu menuItems={defaultMenuItems} translate={true}>
      <LanguageSelect supportedLanguages={Object.values(SUPPORTED_LANGUAGES)} />
    </Menu>
  );
};

MenuWithLanguageSelect.story = {
  name: 'Menu with language selection'
};
