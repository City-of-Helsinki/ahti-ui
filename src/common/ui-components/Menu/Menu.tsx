import React, { useState } from 'react';
import { Link as RouterLink, useHistory } from 'react-router-dom';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';
import { TFunction } from 'i18next';

import Search from '../Search/Search';
import LanguageSelect from '../LanguageSelect/LanguageSelect';
import { SUPPORTED_LANGUAGES } from '../../../common/translation/TranslationConstants';
import { useOvermind } from '../../../domain/overmind';
import AhtiLogo from '../AhtiLogo/AhtiLogo';
import MenuIcon from '../MenuIcon/MenuIcon';
import NavDropdown from './NavDropdown/NavDropdown';
import styles from './Menu.module.scss';

export type MenuItem = {
  readonly categoryIds: string[];
  readonly tagIds: string[];
  readonly name: string;
  readonly url?: string;
  readonly outbound?: boolean;
  readonly disabled?: boolean;
};

export type MenuCategory = {
  readonly title: string;
  readonly category: string;
  readonly menuItems: MenuItem[];
};

export interface MenuProps {
  readonly translate?: boolean;
  readonly menuDark?: boolean;
  readonly menuCategories: MenuCategory[];
  onSelect?(menuItem: MenuItem): void;
  onLogoClick?(): void;
}

const translateMenuCategories = (
  t: TFunction,
  menuCategories: MenuCategory[]
): MenuCategory[] => {
  return menuCategories.map(
    (menuCategory: MenuCategory): MenuCategory => {
      return {
        ...menuCategory,
        title: t(`menu.title.${menuCategory.title}`),
        menuItems: menuCategory.menuItems.map(
          (menuItem: MenuItem): MenuItem => {
            return {
              ...menuItem,
              name: t(`menu.link.${menuCategory.title}.${menuItem.name}`),
            };
          }
        ),
      };
    }
  );
};

const Menu: React.FC<MenuProps> = ({
  translate = false,
  menuDark,
  menuCategories,
  onSelect,
  onLogoClick,
}) => {
  const { state, actions } = useOvermind();
  const [isOpen, setIsOpen] = useState(false);
  const history = useHistory();
  const { t } = useTranslation();

  const items: MenuCategory[] = translate
    ? translateMenuCategories(t, menuCategories)
    : menuCategories;

  const renderMenuContent = () => {
    return (
      <div className={styles.menuContentContainer}>
        {items.map((item: MenuCategory, id: number) => {
          return (
            <NavDropdown key={id} title={item.title} category={item.category}>
              <nav className={styles.navDropdownContent}>
                {item.menuItems.map((menuItem: MenuItem, id: number) => {
                  return (
                    <button
                      key={id}
                      className={classNames(
                        styles.navLink,
                        menuItem.disabled && styles.navLinkDisabled
                      )}
                      disabled={menuItem.disabled}
                      onClick={() => {
                        setIsOpen(false);
                        onSelect && onSelect(menuItem);
                      }}
                      aria-label={menuItem.name}
                      tabIndex={0}
                    >
                      {menuItem.name}
                    </button>
                  );
                })}
              </nav>
            </NavDropdown>
          );
        })}
      </div>
    );
  };

  return (
    <div
      className={
        isOpen ? classNames(styles.containerOpen) : classNames(styles.container)
      }
    >
      <div className={styles.headerContainer}>
        <div className={styles.menuElement}>
          <RouterLink to={'/'} onClick={() => onLogoClick && onLogoClick()}>
            <AhtiLogo fillColor={menuDark || isOpen ? '#001A33' : 'white'} />
          </RouterLink>
        </div>
        <div className={styles.menuElement}>
          <div>
            <LanguageSelect
              supportedLanguages={Object.values(SUPPORTED_LANGUAGES)}
              darkMenu={menuDark}
            />
          </div>
        </div>
        <div
          className={styles.menuElementFront}
          onClick={() => setIsOpen(true)}
        >
          <Search
            featuresToSearch={state.features}
            onSelect={(ahtiId) => {
              history.push('/content');
              actions.selectFeatureById(ahtiId);
            }}
            isMenuOpen={isOpen}
            isDark={menuDark}
          />
        </div>
        <div className={styles.menuElement}>
          <button
            className={styles.toggleMenuButton}
            onClick={() => setIsOpen(!isOpen)}
            aria-label={isOpen ? t('menu.close') : t('menu.open')}
            tabIndex={0}
          >
            {isOpen ? (
              <MenuIcon isDark={menuDark} isOpen={true} />
            ) : (
              <MenuIcon isDark={menuDark} />
            )}
          </button>
        </div>
      </div>
      {isOpen && renderMenuContent()}
    </div>
  );
};

export default Menu;
