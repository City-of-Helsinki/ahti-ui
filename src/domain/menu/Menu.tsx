import React, { useState } from 'react';
import classNames from 'classnames/bind';
import { useTranslation } from 'react-i18next';
import { ReactComponent as AhtiLogo } from '../../assets/icons/ahti_logo.svg';
import styles from './Menu.module.scss';
import NavDropdown from './NavDropdown';
import { useLocation } from 'react-router';
import { Link as RouterLink } from 'react-router-dom';

import { IconLocation, IconFood, IconMenu, IconClose } from 'hds-react';

const cx = classNames.bind(styles);

export type Link = {
  readonly name: string;
  readonly url: string;
  readonly outbound?: boolean;
  readonly disabled?: boolean;
};

export type MenuItem = {
  readonly title: string;
  readonly icon: React.ReactElement;
  readonly links: Link[];
};

export interface MenuProps {
  readonly menuItems?: MenuItem[];
}

export const defaultMenuItems: MenuItem[] = [
  {
    title: 'islands',
    icon: <IconLocation />,
    links: [{ name: 'all_islands', url: '/map?type=myhelsinki' }],
  },
  {
    title: 'travel_to_islands',
    icon: <IconLocation />,
    links: [
      { name: 'water_bus_and_ferries', url: '/', disabled: true },
      { name: 'boat_rides_and_taxis', url: '/', disabled: true },
      { name: 'sightseeing_and_cruises', url: '/', disabled: true },
    ],
  },
  {
    title: 'saunas_and_swimming',
    icon: <IconLocation />,
    links: [
      { name: 'saunas', url: '/', disabled: true },
      { name: 'swimming', url: '/', disabled: true },
      { name: 'ice_swimming', url: '/', disabled: true },
    ],
  },
  {
    title: 'restaurants',
    icon: <IconFood />,
    links: [
      { name: 'restaurants', url: '/', disabled: true },
      { name: 'cafes', url: '/', disabled: true },
      { name: 'beer_and_wine_bars', url: '/', disabled: true },
    ],
  },
  {
    title: 'boater_services',
    icon: <IconLocation />,
    links: [
      { name: 'mooring', url: '/', disabled: true },
      { name: 'maintenance', url: '/', disabled: true },
      { name: 'septic_tank_draining', url: '/', disabled: true },
    ],
  },
  {
    title: 'rental_boats',
    icon: <IconLocation />,
    links: [
      { name: 'rowboats', url: '/', disabled: true },
      { name: 'motorboats', url: '/', disabled: true },
      { name: 'other', url: '/', disabled: true },
    ],
  },
];

const Menu: React.FC<MenuProps> = ({ menuItems = defaultMenuItems }) => {
  const [isOpen, setIsOpen] = useState(false);
  const { t, i18n } = useTranslation();
  const location = useLocation();

  const translatedMenuItems = menuItems.map(
    (menuItem: MenuItem): MenuItem => {
      return {
        ...menuItem,
        title: t(`menu.title.${menuItem.title}`),
        links: menuItem.links.map(
          (link: Link): Link => {
            return {
              ...link,
              name: t(`menu.link.${menuItem.title}.${link.name}`),
            };
          }
        ),
      };
    }
  );

  const renderLanguageButtons = () => {
    // TODO: Investigate: Can't use i18n.languages here as changeLanguage mutates
    //  the initial ['fi', 'en'] list to ['en'] when english is selected.
    return (
      <div>
        {['fi', 'en'].map((language: string, id: number) => {
          return (
            <button
              key={id}
              onClick={() => i18n.changeLanguage(language)}
              className={cx({
                languageButton: true,
                languageButtonEnabled: language === i18n.language,
              })}
            >
              {language}
            </button>
          );
        })}
      </div>
    );
  };

  const renderMenuContent = () => {
    return (
      <div className={styles.menuContentContainer}>
        {translatedMenuItems.map((item: MenuItem, id: number) => {
          return (
            <NavDropdown key={id} title={item.title} icon={item.icon}>
              <nav className={styles.navDropdownContent}>
                {item.links.map((link: Link, id: number) => {
                  return (
                    <RouterLink
                      key={id}
                      to={link.url}
                      className={cx({
                        navLink: true,
                        navLinkDisabled: link.disabled,
                      })}
                    >
                      {link.name}
                    </RouterLink>
                  );
                })}
              </nav>
            </NavDropdown>
          );
        })}
      </div>
    );
  };

  return (
    <div
      className={cx({
        container: true,
        mapPageMenu: location.pathname === '/map',
      })}
    >
      <div className={styles.headerContainer}>
        <RouterLink to={'/'}>
          <AhtiLogo />
        </RouterLink>
        {isOpen && renderLanguageButtons()}
        <div onClick={() => setIsOpen(!isOpen)}>
          {isOpen ? (
            <IconClose className={styles.icon} />
          ) : (
            <IconMenu className={styles.icon} />
          )}
        </div>
      </div>
      {isOpen && renderMenuContent()}
    </div>
  );
};

export default Menu;
