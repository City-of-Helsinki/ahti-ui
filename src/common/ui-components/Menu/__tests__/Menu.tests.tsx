import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

import Menu, { MenuCategory } from '../Menu';
import NavDropdown from '../NavDropdown/NavDropdown';

describe('Menu', () => {
  const menuItems: MenuCategory[] = [
    {
      title: 'islands',
      category: 'ahti:category:island',
      menuItems: [{ id: '', name: 'all_islands', url: '/map?type=myhelsinki' }]
    },
    {
      title: 'restaurants',
      category: 'ahti:category:island',
      menuItems: [
        { id: '', name: 'restaurants', url: '/', disabled: true },
        { id: '', name: 'cafes', url: '/', disabled: true },
        { id: '', name: 'beer_and_wine_bars', url: '/', disabled: true }
      ]
    }
  ];
  const wrapper = mount(
    <MemoryRouter>
      <Menu menuCategories={menuItems} />
    </MemoryRouter>
  );

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('opens menu on click', () => {
    wrapper.find('.toggleMenuButton').simulate('click');
    expect(wrapper.find(NavDropdown)).toHaveLength(2);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
