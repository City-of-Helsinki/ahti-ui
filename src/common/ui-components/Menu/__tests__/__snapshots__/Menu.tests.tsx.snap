// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu opens menu on click 1`] = `"<div class=\\"container\\"><div class=\\"headerContainer\\"><a href=\\"/\\"><svg>ahti_logo.svg</svg></a><button class=\\"toggleMenuButton\\" aria-label=\\"close menu\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 15 15\\" class=\\"icon\\"><path fill-rule=\\"nonzero\\" d=\\"M13.412 0L15 1.588 9.088 7.5 15 13.412 13.412 15 7.5 9.088 1.588 15 0 13.412 5.911 7.5 0 1.588 1.588 0 7.5 5.912 13.412 0z\\"></path></svg></button></div><div class=\\"menuContentContainer\\"><div class=\\"container\\"><div class=\\"headerContainer\\" role=\\"button\\" aria-pressed=\\"false\\" aria-label=\\"open submenu\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 23 23\\" class=\\"icon\\"><path d=\\"M11.6911445,6.57252031e-13 C13.9380141,6.57252031e-13 16.1856662,0.848347826 17.8995793,2.54582609 C21.3274054,5.94 21.0957532,11.2296522 17.8995793,14.8382609 C14.618101,18.5446957 12.868188,20.8565217 11.6911445,23.1793043 C10.5203619,20.8565217 8.7657532,18.5446957 5.48505755,14.8382609 C2.28888364,11.2296522 2.05644886,5.94 5.48505755,2.54582609 C7.19897059,0.848347826 9.44584016,6.57252031e-13 11.6911445,6.57252031e-13 Z M11.6911445,1.73895652 C9.80818799,1.73895652 8.03792712,2.46443478 6.70827494,3.78078261 C5.39505755,5.08069565 4.71418799,6.74295652 4.73923146,8.58913043 C4.76349233,10.4126087 5.49053581,12.222 6.78653581,13.6862609 C9.03966625,16.2305217 10.568101,18.1166087 11.6927097,19.7773043 C12.8220141,18.111913 14.348101,16.2266087 16.598101,13.6862609 C17.894101,12.222 18.6211445,10.4126087 18.6454054,8.58913043 C18.6688836,6.74295652 17.9880141,5.08069565 16.6763619,3.78078261 C15.3451445,2.46443478 13.5756662,1.73895652 11.6911445,1.73895652 Z M11.6922402,5.21538261 C13.6127619,5.21538261 15.168588,6.7719913 15.168588,8.69251304 C15.168588,10.6122522 13.6127619,12.1688609 11.6922402,12.1688609 C9.77171842,12.1688609 8.21510973,10.6122522 8.21510973,8.69251304 C8.21510973,6.7719913 9.77171842,5.21538261 11.6922402,5.21538261 Z M11.6922402,6.95355652 C10.7327619,6.95355652 9.95328364,7.73381739 9.95328364,8.69251304 C9.95328364,9.65042609 10.7327619,10.430687 11.6922402,10.430687 C12.6509358,10.430687 13.4304141,9.65042609 13.4304141,8.69251304 C13.4304141,7.73381739 12.6509358,6.95355652 11.6922402,6.95355652 Z\\"></path></svg><div class=\\"titleContainer\\">islands</div><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"icon angleIconClosed\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></div></div><div class=\\"container\\"><div class=\\"headerContainer\\" role=\\"button\\" aria-pressed=\\"false\\" aria-label=\\"open submenu\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"icon\\"><path d=\\"M13.9714253,10.9514872 C14.1495802,10.7554905 14.4021203,10.6437617 14.6669574,10.6437617 L15.7581797,10.6437617 C16.2917391,10.6437617 16.8001765,10.4170603 17.1567126,10.0201253 L21,5.74180501 L20.4183844,5.16018943 L17.5935184,7.98505543 L17.1781003,7.56963729 L19.9198676,4.66167257 L19.3382897,4.08009472 L16.4303627,6.821862 L16.0149068,6.40648158 L18.8397728,3.58161558 L18.258195,3 L13.9798747,6.84324964 C13.5829397,7.19982348 13.3562006,7.70826089 13.3562006,8.24182028 L13.3562006,9.4018829 C13.3562006,9.66513571 13.2458298,9.91628017 13.0519453,10.0943219 L11.7379445,11.309155 L3.64251505,3.18784912 C3.39091794,2.93621428 2.96044924,3.14269743 3.00292275,3.49606501 C3.24086497,5.47538366 4.30553191,7.76171113 6.54153997,9.99771918 L8.18653998,11.6427569 L9.95854562,11.6427569 L10.5907069,12.3698613 L4.03439548,18.4314657 L5.65114268,20.0338036 L11.6041869,13.5355065 L17.3328306,20.1243333 L18.9900521,18.5844608 L12.7310553,12.305472 L13.9714253,10.9514872 Z\\"></path></svg><div class=\\"titleContainer\\">restaurants</div><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 199.404 199.404\\" class=\\"icon angleIconClosed\\"><g><polygon points=\\"63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  \\"></polygon></g></svg></div></div></div></div>"`;

exports[`Menu renders the component 1`] = `"<div class=\\"container\\"><div class=\\"headerContainer\\"><a href=\\"/\\"><svg>ahti_logo.svg</svg></a><button class=\\"toggleMenuButton\\" aria-label=\\"open menu\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" class=\\"icon\\"><path d=\\"M1.01301075,4.01419355 L22.9875806,4.01419355 L22.9875806,6.61634409 L1.01301075,6.61634409 L1.01301075,4.01419355 Z M1.01301075,17.7482258 L22.9875806,17.7482258 L22.9875806,20.3503763 L1.01301075,20.3503763 L1.01301075,17.7482258 Z M1.01301075,10.8815054 L22.9875806,10.8815054 L22.9875806,13.4836559 L1.01301075,13.4836559 L1.01301075,10.8815054 Z\\"></path></svg></button></div></div>"`;
